### PR TITLE
Update meta-resin submodule and use new flasher strategy

### PR DESCRIPTION
**Pull request by floion**
@agherzan @telphan

### CODE REVIEW

**Comment by telphan** [comment]
LGTM

**Comment by telphan** [comment]
LGTM

File: layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/uEnv.txt_internal [hunk 1]
```diff
@@ -0,0 +1,4 @@
+setemmcroot=if mmc dev 0 ; then setenv mmcroot /dev/mmcblk1p2 rw ; else setenv mmcroot /dev/mmcblk0p2 rw ; fi;
```
Comment (agherzan) [inline layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/uEnv.txt_internal]: This needs some cleanup as it reintroduces dead variables:
https://github.com/resin-os/meta-resin/commit/22a811d1763e736b75ed05f826724ec5d48846a5

File: layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging_%.bbappend [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
-    file://0001-ti_armv7_common.h-Don-t-hardcode-dev-0-but-use-mmcde.patch"
+    file://0001-ti_armv7_common.h-Don-t-hardcode-dev-0-but-use-mmcde.patch \
+    file://uEnv.txt_internal \
+    "
 
 SRC_URI_append_beaglebone-green-wifi = " \
     file://0003-am335x_evm.h-Change-default-dtb-for-Beaglebone-Green-Wireless.patch"
+
+do_deploy_append () {
+        install ${WORKDIR}/uEnv.txt_internal ${DEPLOYDIR}
```
Comment (agherzan) [inline layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging_%.bbappend]: Use the same number of spaces per tabs as above.

File: layers/meta-resin [hunk 3]
```diff
@@ -1 +1 @@
-Subproject commit 3d559030095ab6e1b900b6e3581139c7d7fabf6c
```
Comment (agherzan) [inline layers/meta-resin]: NIT: "Update meta-resin to master" > "meta-resin: Update master to HEAD"

File: layers/meta-resin-beaglebone/recipes-core/images/resin-image-flasher.bbappend [hunk 4]
```diff
@@ -1 +1,3 @@
 include resin-image.inc
+
+RESIN_BOOT_PARTITION_FILES_append_beaglebone = " uEnv.txt_internal: "
```
Comment (agherzan) [inline layers/meta-resin-beaglebone/recipes-core/images/resin-image-flasher.bbappend]: NIT: Append only needs a space before string.

Comment (floion) [inline layers/meta-resin-beaglebone/recipes-core/images/resin-image-flasher.bbappend]: That space at the end was not added for the _append. It was added for clarity on the RESIN_BOOT_PARTITION_FILES variable itself. Changed as per the comment however

File: layers/meta-resin (hunk 3 above)
Comment (floion) [inline layers/meta-resin]: Done

File: layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/uEnv.txt_internal (hunk 1 above)
Comment (floion) [inline layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/uEnv.txt_internal]: Done

File: layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging_%.bbappend (hunk 2 above)
Comment (floion) [inline layers/meta-resin-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging_%.bbappend]: Done

**Comment by agherzan** [comment]
LGTM

**Comment by agherzan** [comment]
Good to merge when build succeeds.

**Pull request merged**